openupgradelib/openupgrade_merge_records.py

```diff
@@ -418,26 +419,52 @@ def _change_generic(env, model_name, record_ids, target_record_id,
                 (model_column, '=', model_name),
                 (res_id_column, 'in', record_ids)])
             if records:
-                records.write({res_id_column: target_record_id})
+                if model_to_replace != 'mail.followers':
+                    records.write({res_id_column: target_record_id})
+                else:
+                    # We need to avoid duplicated results in this model
+                    target_duplicated = model.search([
+                        (model_column, '=', model_name),
+                        (res_id_column, '=', target_record_id),
+                        ('partner_id', 'in', records.mapped('partner_id').ids),
+                    ])
+                    duplicated = records.filtered(lambda x: (
+                        x.partner_id in target_duplicated.mapped('partner_id'))
+                    )
```

Comment:
:hand: Beware bottleneck ahead! Change to:
```suggestion
                    target_duplicated_partners = target_duplicated.mapped('partner_id')
                    duplicated = records.filtered(lambda x: (
                        x.partner_id in target_duplicated_pids)
                    )
```

Reply:
I think with previous code it gets prefetched and no penalty happens, but no problem in changing it.

Reply:
Changed

Reply:
It's not the prefetched, it's because you're ussing mapped instead of filtered, so that's exponential complexity.